src/web/mod.rs

```diff
@@ -141,18 +141,35 @@ impl CratesfyiHandler {
 
 impl Handler for CratesfyiHandler {
     fn handle(&self, req: &mut Request) -> IronResult<Response> {
+        let mut response_time = metrics::ResponseTimer::new();
+        let mut response_route = metrics::ResponseRouter::new();
+
         // try serving shared rustdoc resources first, then router, then db/static file handler
         // return 404 if none of them return Ok
         self.shared_resource_handler
             .handle(req)
-            .or_else(|e| self.router_handler.handle(req).or(Err(e)))
+            .or_else(|e| {
+                let router = self.router_handler.handle(req).or(Err(e));
+
+                response_time.route("router");
+                response_route.route("router");
+                router
+            })
             .or_else(|e| {
                 // if router fails try to serve files from database first
-                self.database_file_handler.handle(req).or(Err(e))
+                let rustdoc = self.database_file_handler.handle(req).or(Err(e));
+
+                response_time.route("docs");
+                response_route.route("docs");
```

Comment:
Just having 'docs' isn't super useful, could you use `DebugPath(&req.url()).to_string()` here instead? If you want to have the metadata that it's a doc path maybe you could add that as a separate counter.

Reply:
You don't want to split the metrics up too finely, aggregating all paths that result in returning a file from S3 makes sense to me. One way to improve this could be to actually inspect the response and subset these like `docs (found)`, `docs (redirect)` and `docs (404)` so that these individual categories can be tracked (e.g. in case a change makes calculating redirects much more expensive).